src/main/java/com/pnudev/springonlinedelivery/config/AppConfig.java

```diff
@@ -0,0 +1,34 @@
+package com.pnudev.springonlinedelivery.config;
+
+import com.pnudev.springonlinedelivery.repos.UserRepository;
+import lombok.RequiredArgsConstructor;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.security.authentication.AuthenticationManager;
+import org.springframework.security.config.annotation.authentication.configuration.AuthenticationConfiguration;
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.security.core.userdetails.UsernameNotFoundException;
+import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
+import org.springframework.security.crypto.password.PasswordEncoder;
+
+@Configuration
+@RequiredArgsConstructor
+public class AppConfig {
+    private final UserRepository userRepository;
+
+    @Bean
+    public UserDetailsService userDetailsService() throws UsernameNotFoundException {
```

Comment:
That's also possible, but better practice would be to implement custom classes that implement `UserDetails` and `UserDetailsService`. That way you wouldn't have to inject a controller inside a bean that is not expected to have anything injected in the first place.